CMakeLists.txt

```diff
@@ -169,37 +170,48 @@ if(NOT DEFINED GTEST_ROOT)
   set(GTEST_ROOT ${SIMPATH})
 endif()
 
-find_package2(PUBLIC FairMQ VERSION 1.4.0)
+if(BUILD_BASEMQ)
+  find_package2(PUBLIC FairMQ VERSION 1.4.0 REQUIRED)
+endif()
+
 find_package2(PUBLIC DDS)
 find_package2(PUBLIC FairLogger VERSION 1.2.0 REQUIRED)
 
-if(NOT DEFINED Boost_NO_SYSTEM_PATHS)
-  Set(Boost_NO_SYSTEM_PATHS TRUE)
-endif()
-if(NOT DEFINED Boost_NO_BOOST_CMAKE AND CMAKE_VERSION VERSION_LESS 3.15)
-  Set(Boost_NO_BOOST_CMAKE TRUE)
-endif()
-if(Boost_NO_BOOST_CMAKE)
-  # If an older version of boost is found both of the variables below are
-  # cached and in a second cmake run, a good boost version is found even
-  # if the version is to old.
-  # To overcome this problem both variables are cleared before checking
-  # for boost.
-  Unset(Boost_INCLUDE_DIR CACHE)
-  Unset(Boost_LIBRARY_DIRS CACHE)
-endif()
+if(BUILD_BASEMQ OR BUILD_UNITTESTS OR BUILD_EXAMPLES)
+  if(NOT DEFINED Boost_NO_SYSTEM_PATHS)
+    Set(Boost_NO_SYSTEM_PATHS TRUE)
+  endif()
+  if(NOT DEFINED Boost_NO_BOOST_CMAKE AND CMAKE_VERSION VERSION_LESS 3.15)
+    Set(Boost_NO_BOOST_CMAKE TRUE)
+  endif()
+  if(Boost_NO_BOOST_CMAKE)
+    # If an older version of boost is found both of the variables below are
+    # cached and in a second cmake run, a good boost version is found even
+    # if the version is to old.
+    # To overcome this problem both variables are cleared before checking
+    # for boost.
+    Unset(Boost_INCLUDE_DIR CACHE)
+    Unset(Boost_LIBRARY_DIRS CACHE)
+  endif()
```

Comment:
Could we move this boost package finding setup into some `cmake/private/` thing like fairroot_prepare_boost_find_packege()`?

Reply:
We should rather clean it up. E.g.

```
  if(NOT DEFINED Boost_NO_BOOST_CMAKE AND CMAKE_VERSION VERSION_LESS 3.15)
    Set(Boost_NO_BOOST_CMAKE TRUE)
  endif()
```
can go away, since `dev` requires CMake 3.15+

```
  if(NOT DEFINED Boost_NO_SYSTEM_PATHS)
    Set(Boost_NO_SYSTEM_PATHS TRUE)
  endif()
```
is anyways very debatable

```
  if(Boost_NO_BOOST_CMAKE)
    # If an older version of boost is found both of the variables below are
    # cached and in a second cmake run, a good boost version is found even
    # if the version is to old.
    # To overcome this problem both variables are cleared before checking
    # for boost.
    Unset(Boost_INCLUDE_DIR CACHE)
    Unset(Boost_LIBRARY_DIRS CACHE)
  endif()
```
we should check, if this is really needed still with the supported Boost versions - it also sounds as it came around the 1.69 to 1.70 releases (+ the current CMake versions at that time, pre 3.15) when Boost started to ship its own CMake package.

Reply:
partly addressed in 1cadcbac4672d684861f87bc680dac2879c75fbb in #1252